pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -401,13 +401,17 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		existingPods := nodeInfo.PodsWithAffinity()
 		klog.V(6).Infof("Node %v, Pods with Affinity: %v", nodeInfo.Node().Name, len(existingPods))
 		for _, existingPod := range existingPods {
+			// in current arktos perf runs, this should not be reached
+			klog.V(2).Infof("DEBUG: this should not be reached. print out the pod: %v-%v %v", existingPod.Namespace, existingPod.Name)
 			existingPodTopologyMaps, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, existingPod, node)
 			if err != nil {
 				catchError(err)
 				cancel()
 				return
 			}
-			appendTopologyPairsMaps(existingPodTopologyMaps)
+			if existingPodTopologyMaps!=nil {
```

Comment:
what if the initial map is nil/empty?

Reply:
this should end up with empty topologyMap, sicne the appendTopologyPairsMap is not called.

Reply:
my concern is, normally a map starts as the empty one (unless otherwise initialized), which will make the code never be called. not sure whether that is the right intention.

Reply:
this is exactly the code change intended to do. i.e only call this function appendTopologyPairsMaps() when the existingPodTopologyMaps returned from the previous function is not nil, to avoid unnecessary calls, i.e. save a lock-unlock OP per existing POD in the cluster -- this could be huge considering there are 450K pods in the end for a 15k node run.

Reply:
ok got it. thanks for the explanation.